### PR TITLE
PaulDFoster/issue35

### PR DESCRIPTION
**Pull request by PaulDFoster**
#35 Implements:
$Home/.rover as home directory for:
.rover.yaml
TF_DATA_DIR
tfstates

Does not implement for ci_tasks folder, logging or comand history

Tested with terraform cycle.

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -44,3 +45,23 @@ func CopyFile(src string, dest string) error {
...
 }
+
+func GetHomeDirectory() (string, error) {
+	home, err := os.UserHomeDir()
+
+	if err != nil {
+		return "", errors.New("Unable to access user home directory")
+	}
+
+	home += "/.rover"
```
Comment (sebastus) [inline pkg/utils/utils.go:56]: preferable to use filepath.Join(...)

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -44,3 +45,23 @@ func CopyFile(src string, dest string) error {
...
+
+	_, direrr := os.Stat(home)
```
Comment (sebastus) [inline pkg/utils/utils.go:58]: you could do an os.MkdirAll(...) and eliminate several lines. (i.e. no need to check for existence, etc)

File: cmd/root.go [hunk 3]
```diff
@@ -51,22 +52,23 @@ func init() {
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
+
+	home, err := utils.GetHomeDirectory()
+	cobra.CheckErr(err)
+
 	if cfgFile != "" {
 		// Use config file from the flag.
+		console.Info("Use config file from the flag")
```
Comment (sebastus) [inline cmd/root.go:61]: console.Debug ?

File: cmd/root.go [hunk 4]
```diff
@@ -51,22 +52,23 @@ func init() {
...
-
 		// Search config in home directory and CWD with name ".rover" (without extension).
 		viper.AddConfigPath(home)
 		viper.AddConfigPath(".")
 		viper.SetConfigType("yaml")
 		viper.SetConfigName(".rover")
 
 		// Config defaults
-		viper.SetDefault("tempDir", "/tmp")
+		viper.SetDefault("tempDir", home+"/tmp") // Modify to be $home/.rover/tmp
```
Comment (sebastus) [inline cmd/root.go:71]: filepath.Join

File: cmd/root.go [hunk 5]
```diff
@@ -79,7 +81,7 @@ func initConfig() {
 		console.Infof("Using config file: %s\n", viper.ConfigFileUsed())
 	} else {
 		// Fall back to creating empty config file
-		fileName := "./.rover.yaml"
+		fileName := home + "/.rover.yaml" // Modify to be home/.rover/.rover.yaml
```
Comment (sebastus) [inline cmd/root.go:84]: filepath.Join

**Review by sebastus — CHANGES_REQUESTED** [review]
A few little things...

File: cmd/root.go (hunk 3 above)
Comment (PaulDFoster) [reply cmd/root.go:62]: Not my code, so I don't know the intention.

**Review by sebastus — APPROVED** [review]
lgtm

**Pull request merged**